File: pc_shop_backend/src/main/java/com/example/pc_shop_backend/model/Pc.java
package com.example.pc_shop_backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Pc {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String pc_name;

    @Column(precision = 10, scale = 2)
    private BigDecimal base_price;

    private Integer ram;

    private String processor;

    private Integer pc_memory;

    private String image_path;
}

File: pc_shop_frontend/src/models/Pc.ts
export default class Pc {
  id: Number
  pc_name: String
  base_price: Number
  ram: Number
  processor: String
  pc_memory: Number
  image_path: String

  constructor(data: any) {
    this.id = data.id
    this.pc_name = data.pc_name
    this.base_price = data.base_price
    this.ram = data.ram
    this.processor = data.processor
    this.pc_memory = data.pc_memory
    this.image_path = data.image_path
  }
}
